modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/model/listener/ObjectDefinitionModelListener.java

```diff
@@ -0,0 +1,181 @@
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectEntryService;
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.ModelListener;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.Localization;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.workflow.WorkflowHandler;
+import com.liferay.portal.kernel.workflow.WorkflowHandlerRegistryUtil;
+import com.liferay.portal.workflow.kaleo.model.KaleoInstance;
+import com.liferay.portal.workflow.kaleo.service.persistence.KaleoInstanceUtil;
+import com.liferay.portal.workflow.metrics.search.index.InstanceWorkflowMetricsIndexer;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Pedro Leite
+ */
+@Component(service = ModelListener.class)
+public class ObjectDefinitionModelListener
+	extends BaseModelListener<ObjectDefinition> {
+
+	@Override
+	public void onAfterUpdate(
+			ObjectDefinition originalModel, ObjectDefinition model)
+		throws ModelListenerException {
+
+		if (originalModel.getTitleObjectFieldId() !=
+				model.getTitleObjectFieldId()) {
+
+			ObjectDefinition objectDefinition =
+				_objectDefinitionLocalService.fetchObjectDefinition(
+					model.getObjectDefinitionId());
+
+			for (ObjectEntry objectEntry :
+					_objectEntryService.getObjectEntries(
+						objectDefinition.getObjectDefinitionId())) {
+
+				KaleoInstance kaleoInstance =
+					KaleoInstanceUtil.fetchByCN_CPK_First(
+						objectDefinition.getClassName(),
+						objectEntry.getObjectEntryId(), null);
+
+				if (kaleoInstance != null) {
+					try {
+						String objectEntryTitleValue =
+							objectEntry.getTitleValue();
+
+						if (Validator.isNotNull(objectEntryTitleValue)) {
+							_instanceWorkflowMetricsIndexer.updateInstance(
+								kaleoInstance.isActive(),
+								_createAssetTitleLocalizationMap(
+									kaleoInstance.getClassName(),
+									kaleoInstance.getClassPK(),
+									kaleoInstance.getGroupId(),
+									objectEntryTitleValue),
+								objectDefinition.getLabelMap(),
+								kaleoInstance.getCompanyId(),
+								kaleoInstance.getKaleoInstanceId(),
+								kaleoInstance.getModifiedDate());
+						}
+					}
+					catch (PortalException portalException) {
+						throw new ModelListenerException(
+							portalException.getMessage());
+					}
+				}
+			}
+		}
+	}
+
+	private Map<Locale, String> _createAssetTitleLocalizationMap(
```

Comment:
Hey, I believe we don't need this method. Could you please add a Reference for `_indexerHelper` and check if we can use the `_indexerHelper#createAssetTitleLocalizationMap` method in this case?